src/App.tsx

```diff
@@ -1,14 +1,44 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [loader, setLoader] = useState(true);
+  const [selectedOption, setSelectedOption] = useState('all');
+  const [searchInputValue, setSearchInputValue] = useState('');
+
+  useEffect(() => {
+    getTodos().then(fetchedTodos => {
+      const filteredTodos = fetchedTodos.filter(todo =>
+        todo.title.toLowerCase().includes(searchInputValue.toLowerCase()),
+      );
+
+      switch (selectedOption) {
+        case 'all':
+          setTodos(filteredTodos);
+          break;
+        case 'active':
+          setTodos(filteredTodos.filter(todo => !todo.completed));
+          break;
+        case 'completed':
+          setTodos(filteredTodos.filter(todo => todo.completed));
+          break;
+      }
+
+      setLoader(false);
+    });
+  }, [selectedOption, searchInputValue]);
```

Comment:
there is no need to refetch options on every input change. You can fetch todos once and then filter it based on searchValue and store result of filtering in a common variable